lxd/db.go

```diff
@@ -657,71 +658,55 @@ func initDb(d *Daemon) (err error) {
 	return err
 }
 
-var NoSuchImageError = errors.New("No such image")
-
-func dbImageGet(d *Daemon, name string, public bool) (*shared.ImageBaseInfo, error) {
-
-	// count potential images first, if more than one
-	// return error
-	var countImg int
+// Get an ImageBaseInfo object from the database.
+// There can never be more than one image with a given fingerprint, as it is
+// enforced by a UNIQUE constraint in the schema.
+func dbImageGet(db *sql.DB, fingerprint string, public bool) (*shared.ImageBaseInfo, error) {
 	var err error
-	q := "SELECT count(id) FROM images WHERE fingerprint like ?"
-	if public {
-		q = q + " AND public=1"
-	}
+	var create, expire, upload *time.Time // These hold the db-returned times
 
-	arg1 := []interface{}{name + "%"}
-	arg2 := []interface{}{&countImg}
-	err = shared.DbQueryRowScan(d.db, q, arg1, arg2)
-	if err != nil {
-		return nil, err
-	}
+	// The object we'll actually return
+	image := new(shared.ImageBaseInfo)
 
-	if countImg > 1 {
-		return nil, fmt.Errorf("Multiple images for fingerprint")
```

Comment:
For the record, that can never happen (there is a UNIQUE constraint on the schema).